OSDOCS-11383: GCP Workload ID webhook (#83682)
openshift/openshift-docs: modules/pod-short-term-auth-gcp-cluster-sa.adoc

----
$ oc patch serviceaccount <service_account_name> -p '{"metadata": {"annotations": {"cloud.google.com/service-account-email": "<service_account_email>"}}}'
----
+
Replace `<service_account_email>` with the email address for the {gcp-short} service account.
+
[TIP]
====
{gcp-short} service account email addresses typically use the format `<service_account_name>@<project_id>.iam.gserviceaccount.com`
====

. Annotate the service account to use the `direct` external credentials configuration injection mode by running the following command:
+
[source,terminal]
----
$ oc patch serviceaccount <service_account_name> -p '{"metadata": {"annotations": {"cloud.google.com/injection-mode": "direct"}}}'
----
+
In this mode, the Workload Identity Federation webhook controller directly generates the {gcp-short} external credentials configuration and injects them into the pod.

. Using the Google Cloud CLI (`gcloud`), specify the permissions for the workload by running the following command:
+
[source,terminal]
----
$ gcloud projects add-iam-policy-binding <project_id> --member "<service_account_email>" --role "projects/<project_id>/roles/<role_for_workload_permissions>"
----
+

> Replace `<role_for_workload_permissions>` with the role that has the permissions the workload requires.

Reviewer: By `role_for_workload_permissions`, could you please explain exactly what kinds of permissions required here? Is it, `IAM Workload Identity Pool Admin`, or something else? Thanks!

Reviewer 2: Actually it is to grant the permissions needed by the workload, should be decided by the workload user.

Author: I can make this more clear so the user is not left wondering about it

Author: I think this might be a little more clear:
```
Replace `<role_for_workload_permissions>` with the role for the workload. 
Specify a role that grants the permissions that your workload requires.
```

====

. Annotate the service account to use the `direct` external credentials configuration injection mode by running the following command:
+
[source,terminal]
----
$ oc patch serviceaccount <service_account_name> -p '{"metadata": {"annotations": {"cloud.google.com/injection-mode": "direct"}}}'
----
+
In this mode, the Workload Identity Federation webhook controller directly generates the {gcp-short} external credentials configuration and injects them into the pod.

. Using the Google Cloud CLI (`gcloud`), specify the permissions for the workload by running the following command:
+
[source,terminal]
----
$ gcloud projects add-iam-policy-binding <project_id> --member "<service_account_email>" --role "projects/<project_id>/roles/<role_for_workload_permissions>"
----
+
Replace `<role_for_workload_permissions>` with the role for the workload. 
Specify a role that grants the permissions that your workload requires.

.Verification

* To verify the service account configuration, inspect the `ServiceAccount` manifest by running the following command:
+
[source,terminal]
----
$ oc get serviceaccount <service_account_name>
----
+

> The following example allows the `service-a/app-x` {product-title} service account to impersonate a {gcp-short} service account called `app-x`:

Reviewer 3: ```suggestion
In the following example, the `service-a/app-x` {product-title} service account can impersonate a {gcp-short} service account called `app-x`:
```
Revised a little to avoid anthropomorphic language ("The following example allows...").
